### PR TITLE
VMT: add KCSA for CVE-2020-2023 and CVE-2020-2026

### PR DESCRIPTION
**Pull request by amshinde**
These have been fixed and put in releases 1.11.1 and 1.10.5.
Sending out announcement.

Signed-off-by: Archana Shinde <archana.m.shinde@intel.com>

### CODE REVIEW

File: VMT/KCSA/KCSA-CVE-2020-2023.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+  - name: `Yuval Avrahami`
+    affiliation: `Palo Alto Networks`
+    reported:
+      - CVE-2020-2023
+
+issues:
+
+  links:
+    - https://github.com/kata-containers/agent/issues/791
```
Comment (YuvalAvra) [inline VMT/KCSA/KCSA-CVE-2020-2023.md:29]: Should probably also include https://github.com/kata-containers/runtime/issues/2476 and https://github.com/kata-containers/runtime/issues/2488. The matching pull requests are https://github.com/kata-containers/runtime/pull/2477 and https://github.com/kata-containers/runtime/pull/2487.

File: VMT/KCSA/KCSA-CVE-2020-2023.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+       the device and now points to garbage data, resulting in a `SIGSEGV`.
+    6. In case of no DAX, similar outcome can be achieved by running a process to 
+       exhaust memory and cause guest kernel to free page cache including pages allocated
+       for the kata-agent binary. The kernel would need to remap the code region 
+       of the kata-agent process to its executable, and read the kata-agent 
+       binary directly from the compromised guest filesystem device. 
+
+notes:
+  - The vulnerability can be used to compromise the guest and masquerade 
+    as the kata-agent. All users are recommended to upgrade to mitigate guest breakout.
```
Comment (YuvalAvra) [inline VMT/KCSA/KCSA-CVE-2020-2023.md:57]: Consider adding:
'
To exploit the issue, the container must possess CAP_MKNOD.
'

File: VMT/KCSA/KCSA-CVE-2020-2023.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    3. Delete the kata-agent binary on the guest.
+    4. Write several big files with garbage data to the device so that the blocks 
+       previously allocated for the kata-agent binary will be used.
+    5. In case of DAX, the kata-agent process code region is directly mapped to 
+       the device and now points to garbage data, resulting in a `SIGSEGV`.
+    6. In case of no DAX, similar outcome can be achieved by running a process to 
+       exhaust memory and cause guest kernel to free page cache including pages allocated
+       for the kata-agent binary. The kernel would need to remap the code region 
+       of the kata-agent process to its executable, and read the kata-agent 
+       binary directly from the compromised guest filesystem device.
```
Comment (YuvalAvra) [inline VMT/KCSA/KCSA-CVE-2020-2023.md:53]: Perhaps explain it a bit more generally:
'
A malicious container can create a device file for the guest root filesystem device, and use it to modify the guest filesystem through utilities like [`debugfs`](https://linux.die.net/man/8/debugfs), potentially allowing a container-to-guest breakout:

1. Find the guest root filesystem device major and minor numbers by inspecting `/sys/dev/block`.
2. Use`mknod` to create a device file for the guest root filesystem device.
3. Use utilities such as `debugfs` to access the device file and modify the guest filesystem.
4. Attempt to gain code execution on the guest by overwriting crucial guest files (e.g. `kata-agent`, `libc`)

When the guest filesystem is mounted with [DAX](https://www.kernel.org/doc/Documentation/filesystems/dax.txt), it's easier for the container to gain guest code execution. With DAX, changes made to the device immediately propagate to the pages used by guest processes. This means the container can inject code to guest processes by modifying the executables and libraries used by them.

Without DAX, the malicious container can force changes made to the device to propagate to guest pages by exhausting memory, forcing the guest kernel to re-read the pages from the compromised device. The attack may fail if the container memory is limited by cgroups.
'

File: VMT/KCSA/KCSA-CVE-2020-2026.md [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+    3. The malicious guest returns a response for CreateSandbox
+    4. Once the kata runtime on the host receives the CreateSandbox response, it tries to 
+       bind mount the container image at /run/kata-containers/shared/sandbox/$sbx_id/${first_ctrid}/rootfs
+    5. The malicious symlink redirects the mount operation to the target on the host.
+
+    In case of non initial container images, the guest does not know the container id.
+    In case of volume mounts and mounts related to platform-specific files such as
+    `/etc/hosts`, the mounts are performed in the shared directory on paths that are not
+    known to the guest. The guest would need to win a race condition between
+    `ensureDestinationExists` function call in the agent that creates the file or
```
Comment (YuvalAvra) [inline VMT/KCSA/KCSA-CVE-2020-2026.md:71]: This function is [called](https://github.com/kata-containers/runtime/blob/ebe9677f23b574c5defacf57456d221d8ce901f2/virtcontainers/mount.go#L276) by the `kata-runtime`.

**Comment by YuvalAvra** [comment]
Made some minor suggestions, other than that LGTM.

File: VMT/KCSA/KCSA-CVE-2020-2023.md (hunk 3 above)
Comment (amshinde) [reply VMT/KCSA/KCSA-CVE-2020-2023.md:53]: @YuvalAvra yes, this looks much better. Will update.

**Comment by amshinde** [comment]
@YuvalAvra Thanks for the review. I have addressed your comments. Please take a look.

**Comment by amshinde** [comment]
@jodh-intel @devimc PTAL

**Pull request merged**